providers/github/provider/provider.go

```diff
@@ -73,19 +80,25 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
 		pk := req.Flags[connection.OPTION_APP_PRIVATE_KEY]
 		conf.Options[connection.OPTION_APP_PRIVATE_KEY] = string(pk.Value)
 		isAppAuth = true
+		log.Debug().Msg("application credentials provided")
 	}
 
 	token := ""
 	if x, ok := flags["token"]; ok && len(x.Value) != 0 {
 		token = string(x.Value)
+		log.Debug().Msg("loaded token from flag")
 	}
-	if token == "" {
+	if token == "" && len(os.Getenv("GITHUB_TOKEN")) != 0 {
 		token = os.Getenv("GITHUB_TOKEN")
+		log.Debug().Msg("loaded token from GITHUB_TOKEN env variable")
 	}
 	if token == "" && !isAppAuth {
 		return nil, errors.New("a valid GitHub authentication is required, pass --token '<yourtoken>', set GITHUB_TOKEN environment variable or provide GitHub App credentials")
 	}
 	if token != "" {
+		if isAppAuth {
+			log.Warn().Msg("both authentication methods provided, using application credentials")
```

Comment:
it looks like here we warn that we're going to use the application credentials but then set the token as the credential? is that correct?

Reply:
That is correct. We can also NOT set it, it is a noop at this time so, more like a preference.

Reply:
ok. i do find it confusing to set it if we're not using it, but things are working 👍

Reply:
Will not set it to avoid confusion. 👍🏽